### DISCUSSION TITLE
Weird issue with Laravel bindings

### DISCUSSION THREAD

**Discussion by BrokenR3C0RD** [laravel/framework#36284] [Q&A] [answered]
Hi there, I wasn't sure if this was a Laravel issue or not but I thought I might as well ask here first before making an issue.
I'm writing a web application for my work, and I'm running into a very odd issue.
Long story short, I'm using the Lighthouse GraphQL framework and I'm writing a custom directive to allow searching of a certain field. The way that the directive works is that it will be passed an Eloquent querybuilder and a value, and will return said querybuilder.

Now for the issue. No matter what I pass into my bindings, `$builder.getBindings()` always shows that the items added to the builder are casted to `true`. I've tried casting them, converting them into strings, and I looked for other issues via Google and had very little luck finding anything related.

Here's the code in question:
```php
    /**
     * Add additional constraints to the builder based on the given argument value.
     *
     * @param \Illuminate\Database\Eloquent\Builder|\Illuminate\Database\Query\Builder $builder
     * @param mixed                                                                    $value
     *
     * @return \Illuminate\Database\Eloquent\Builder|\Illuminate\Database\Query\Builder
     */
    public function handleBuilder($builder, $value)
    {
        $relation = (string) $this->directiveArgValue('relation', $this->nodeName());
        $where = (array) $this->directiveArgValue('where');

        if (null !== $value) {
            foreach ($value as $search) {
                $builder = $builder->whereHas($relation, function (\Illuminate\Database\Eloquent\Builder $query) use ($where, $search) {
                    foreach ($where as $condition) {
                        \Log::debug($condition['key'].' '.$search[$condition['key']]);
                        if (array_key_exists($condition['key'], $search)) {
                            \Log::debug($condition['key'].' '.$search[$condition['key']].' '.gettype($search[$condition['key']]));
                            $key = $condition['key'];
                            $value = $search[$key];
                            $query->where($key, $condition['operator'] || 'LIKE', $value, 'and');
                        } else {
                            \Log::debug("Doesn't exist?");
                        }
                    }
                });
            }

            \Log::debug($builder->toSql());
            \Log::debug(var_export($builder->getBindings(), true));

            return $builder;
        } else {
            return $builder;
        }
    }
```
(yes, the code is a mess, I haven't gotten around to cleaning up the code yet since it isn't even working at the moment)
The output in the logs shows this:
```
[2021-02-16 14:50:59] local.DEBUG: feature_id 1  
[2021-02-16 14:50:59] local.DEBUG: feature_id 1 string  
[2021-02-16 14:50:59] local.DEBUG: value -1  
[2021-02-16 14:50:59] local.DEBUG: value -1 integer  
[2021-02-16 14:50:59] local.DEBUG: select * from "homes" where exists (select * from "home_features" where "homes"."id" = "home_features"."home_id" and "feature_id" = ? and "value" = ?) and "homes"."deleted_at" is null  
[2021-02-16 14:50:59] local.DEBUG: array (
  0 => true,
  1 => true,
)  
```

I hope I can get a bit of insight into this problem if it's an obvious thing I didn't see. Thanks for your time!

**Comment by BrokenR3C0RD** [comment]
I just updated Laravel to v8.28.0 after seeing that there was an advisory about query builder issues (https://github.com/laravel/framework/security/advisories/GHSA-x7p5-p2c9-phvg) but the issue still appears

**Comment by mnabialek** [comment, marked as answer]
For me in:

```
$query->where($key, $condition['operator'] || 'LIKE', $value, 'and');
```

this part `$condition['operator'] || 'LIKE'` is invalid. It probably always returns `true`. Instead it should be probably something like `$condition['operator'] ?? 'LIKE'`
